File: src/main/java/net/slipp/support/web/CorsFilter.java
package net.slipp.support.web;

import javax.servlet.*;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;

public class CorsFilter implements Filter {
    @Override
    public void init(FilterConfig filterConfig) throws ServletException {

    }

    @Override
    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
        HttpServletResponse response = (HttpServletResponse) servletResponse;
        response.setHeader("X-Frame-Options", "ALLOW-FROM https://apps.facebook.com");
        filterChain.doFilter(servletRequest, servletResponse);
    }

    @Override
    public void destroy() {

    }
}

File: src/main/scala/net/slipp/support/web/CorsFilter.scala
package net.slipp.support.web

import javax.servlet._
import javax.servlet.http.{HttpServletRequest, HttpServletResponse}

class CorsFilter extends Filter {
  def init(filterConfig: FilterConfig) {
  }

  def doFilter(servletRequest: ServletRequest, servletResponse: ServletResponse, filterChain: FilterChain) {
    val request = servletRequest.asInstanceOf[HttpServletRequest]
    val response = servletResponse.asInstanceOf[HttpServletResponse]
    val uri = request.getRequestURI

    uri match {
      case "/attachments" => response.setHeader("X-Frame-Options", "SAMEORIGIN")
      case _ => response.setHeader("X-Frame-Options", "ALLOW-FROM https://apps.facebook.com")
    }

    filterChain.doFilter(servletRequest, servletResponse)
  }

  def destroy {
  }
}
